File: ShoppingListProject/ShoppingList.js
const prompt = require("prompt-sync")({ sigint: true });

let shoppingList = [];

console.log("Enter up to 5 items to buy:");
for (let i = 0; i < 5; i++) {
    let item = prompt("> ");
    if (item.trim() === "") break;
    shoppingList.push(item);
}

console.log("\nYour shopping list:");
console.log(shoppingList.join(", "));

console.log(\nYou entered ${shoppingList.length} item(s).);

let searchItem = prompt("\nSearch for an item: ");
if (shoppingList.includes(searchItem)) {
    console.log(${searchItem} is in your shopping list!);
} else {
    console.log(${searchItem} is NOT in your shopping list.);
}

File: ShoppingList.java
import java.util.Scanner;

public class ShoppingList {
    public static void main(String[] args) {
        Scanner scnr = new Scanner(System.in);

        String[] shoppingList = new String[5];
        int itemCount = 0;

        System.out.println("\nEnter the number of items to add to your shopping list (max 5):");

        for (int i = 0; i < 5; i++) {
            System.out.print("Enter item # " + (i + 1) + ": ");
            String item = scnr.nextLine().trim();

            if (item.isEmpty()) {
                break;
            }

            shoppingList[itemCount] = item;
            itemCount++;
        }

        System.out.println("\nYour Shopping List:");
        for (int i = 0; i < itemCount; i++) {
            System.out.println(">> " + shoppingList[i]);
    }

    System.out.print("\nEnter item to search for: ");
        String searchItem = scnr.nextLine().trim();
        
        boolean found = false;
        
        for (int i = 0; i < itemCount; i++) {
            if (shoppingList[i].equalsIgnoreCase(searchItem)) {
                found = true;
                break;
            }
        }
        if (found) {
            System.out.println(searchItem + " is in your shopping list.");
        } else {
            System.out.println(searchItem + " is not in your shopping list.");
        }
        scnr.close();
    }
}
